src/ast/writing/ast_mapping.ts

```diff
@@ -497,7 +521,12 @@ class TupleExpressionWriter extends ASTNodeWriter {
  */
 abstract class SimpleStatementWriter<T extends Statement> extends ASTNodeWriter {
     writeWhole(node: T, writer: ASTWriter): SrcDesc {
-        return [[node, this.writeInner(node, writer)], ";"];
+        const stmtDesc = super.writeWhole(node, writer);
+
+        stmtDesc.unshift(...writeDocs(node.documentation, writer));
+        stmtDesc.push(";");
+
+        return stmtDesc;
```

Comment:
Why not write this as:

```
return [...writeDocs(node.documentation, writer), [node, this.writeInner(node, writer)], ";"];
```
Like in the other cases?

Reply:
We use `unshift()` in few more places, so it looks more common. Also I wanted to make clear that heading `[[node, this.writeInner(node, writer)]]` is also something, that is done like in other places. The lines:
```js
stmtDesc.unshift(...writeDocs(node.documentation, writer));
stmtDesc.push(";");
```
Are indication how the statement desc _**is modified**_ in this particular writer hierarchy member.